Code/Framework/AzFramework/AzFramework/DocumentPropertyEditor/PropertyEditorSystem.h

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#pragma once
+
+#include <AzFramework/DocumentPropertyEditor/PropertyEditorSystemInterface.h>
+
+namespace AZ::DocumentPropertyEditor
+{
+    class PropertyEditorSystem : public PropertyEditorSystemInterface
+    {
+    public:
+        AZ_RTTI(PropertyEditorSystem, "{5DD4E43F-B17C-463E-8D4C-5A1E22DD452D}", PropertyEditorSystemInterface);
+        AZ_CLASS_ALLOCATOR(PropertyEditorSystem, AZ::OSAllocator, 0);
+
+        PropertyEditorSystem();
+        ~PropertyEditorSystem();
+
+        void RegisterNode(NodeMetadata metadata) override;
+        void RegisterPropertyEditor(PropertyEditorMetadata metadata) override;
+        void RegisterAttribute(AttributeMetadata metadata) override;
+
+        const NodeMetadata* FindNode(AZ::Name name) const override;
+        const PropertyEditorMetadata* FindPropertyEditor(AZ::Name name) const override;
+        const AttributeMetadata* FindAttribute(AZ::Name name, const PropertyEditorMetadata* parent) const override;
+        AZ::Name LookupNameFromCrc(AZ::Crc32 crc) const override;
+
+    private:
+        void AddNameToCrcTable(AZ::Name name);
+
+        AZStd::unordered_map<AZ::Crc32, AZ::Name> m_crcToName;
```

Comment:
We should abstract away from the name the fact that the underlying type to ID attributes are Crc32s.
We want to move away from using Crc32 because they aren't a hash function, but a checksum and the algorithm isn't made for uniformly distributed values with different inputs.

How about calling these functions `LookupNameFromAttributeId` and `AddNameToAttributeIdTable`?

Reply:
I don't mind changing this, but for reference this is purely for interop with `EditContext` - are we actively planning on changing the hashing function for all `EditContext` attributes?